sequent_with_notations.ml

```diff
@@ -0,0 +1,96 @@
+open Sequent
+open Notations
+
+type proof_with_notations = {sequent: sequent; notations: notations}
+
+exception Json_exception of string
+
+let get_key d k =
+    let value =
+        try Yojson.Basic.Util.member k d
+        with Yojson.Basic.Util.Type_error (_, _) -> raise (Json_exception ("request body must be a json object"))
+    in
+    if value = `Null
+    then raise (Json_exception ("required argument '" ^ k ^ "' is missing"))
+    else value
```

Comment:
Same code in `proof_with_notations.ml`, move to `notations.ml`?

Reply:
I agree request json handling deserves a refactoring, that I'll do later on.